### ISSUE TITLE
gsutils hangs in warp

### ISSUE THREAD

**Issue by elviskahoro** [warpdotdev/Warp#1105]
### Describe the bug

```text
We are using google cloud and use gsutil to do things like gsutil cp -r <bucketname> . This command completely hangs in warp. It works in regular terminal. Not all of gsutil fails though. gsutil ls works well.
```

### To Reproduce

Steps to reproduce:
2. Click on '...'
3. Scroll down to '...'
4. See error '...'

**Comment by justinborromeo-glean** [comment]
+1 on `gsutil cp` hanging (repro'ed across several machines)

**Comment by alokedesai** [comment]
@justinborromeo-glean Would you able to provide more concrete steps to repro this? We'd love to get to the bottom of this. Thanks so much!

**Comment by rjurney** [comment]
I have this issue too and it is killing me :)

To reproduce this issue:

1. Download the Google SDK CLI package and install it https://cloud.google.com/sdk/docs/install
2. `gsutil init` (or `gsutil auth login` if you have already initialized)
3. `echo "test_file_contents" > /tmp/test.txt`
4. `gsutil mb <random-test-bucket>`
5. `gsutil cp test_file_contents gs://<random-test-bucket>/`
6. Note that it sits there forever.
7. Open another shell, run the same command, watch it work.
8. Shout "Ah ha!"

It did work for me for a while but then it stopped and I didn't think to try normal Terminal for 24 hours and it works there!

**Comment by rjurney** [comment]
Yes, it does. Warp.dev with default zsh included with OS X 12.3 - zsh 5.8 (x86_64-apple-darwin21.0)

**Comment by alokedesai** [comment]
@rjurney Thanks for the extra information! I'm pretty sure we've found a fix for this issue, though we are still in the process of verifying this fixes a few different issues. I should have more updates within the next few days

**Comment by rjurney** [comment]
@alokedesai Thanks, great news! Can you link to the PR?

**Comment by alokedesai** [comment]
Our internal source code is not yet public, so can't link to the PR just yet unfortunately. The tldr; of the issue was that we weren't properly closing file descriptors before starting the shell, which would cause them to be leaked to the child process and occasionally cause processes to hang.

**Comment by rjurney** [comment]
@alokedesai Ok, I see.

I have another issue... is this intentional? git commands never autocomplete. A bunch of them don't. Is this something I should file a ticket for? The paths don't autocomplete and I have to type them in, which is maddening.

**Comment by alokedesai** [comment]
@rjurney  Mind filing another issue with concrete examples where this doesn't work? Definitely not intended, would love to fix them :)

**Comment by rjurney** [comment]
@alokedesai you got it! I said it blocks my Warp usage and... I still use it, but it really makes me mad and I will eventually have to use something else if it happens indefinitely. I am trying the bash from `brew install bash` to see I... oh wait, I am on zsh! I have to check the version... is that supported?

https://github.com/warpdotdev/Warp/issues/1753

**Comment by rjurney** [comment]
Ok, cool I updated it to say I use `zsh 5.8.1 (x86_64-apple-darwin21.0)`.
